src/services/coingecko/api/price.service.ts

```diff
@@ -167,24 +169,39 @@ export class PriceService {
   private parseHistoricalPrices(
     results: HistoricalPriceResponse[],
     addresses: string[],
-    start: number
+    start: number,
+    aggregateBy: 'day' | 'hour' = 'day'
   ): HistoricalPrices {
     const assetPrices = Object.fromEntries(
       addresses.map((address, index) => {
         address = this.addressMapOut(address);
         const result = results[index].prices;
         const prices = {};
         let dayTimestamp = start;
-        for (const key in result) {
-          const value = result[key];
-          const [timestamp, price] = value;
-          if (timestamp > dayTimestamp * 1000) {
+        if (aggregateBy === 'hour') {
+          const pricesByHour = groupBy(result, r =>
+            getUnixTime(startOfHour(fromUnixTime(r[0] / 1000)))
+          );
+          for (const key of Object.keys(pricesByHour)) {
+            const price = (last(pricesByHour[key]) || [])[1];
             // TODO - remove this conditional once coingecko supports wstETH
-            prices[dayTimestamp * 1000] =
+            prices[Number(key) * 1000] =
               address === this.appAddresses.stETH
                 ? price * TOKENS.Prices.ExchangeRates.wstETH.stETH
                 : price;
-            dayTimestamp += twentyFourHoursInSecs;
+          }
+        } else if (aggregateBy === 'day') {
+          for (const key in result) {
+            const value = result[key];
+            const [timestamp, price] = value;
+            if (timestamp > dayTimestamp * 1000) {
+              // TODO - remove this conditional once coingecko supports wstETH
+              prices[dayTimestamp * 1000] =
+                address === this.appAddresses.stETH
+                  ? price * TOKENS.Prices.ExchangeRates.wstETH.stETH
+                  : price;
```

Comment:
I think it would be nice to extract the logic for `stETH` to a function and reuse it (on lines 189 and 200)

Reply:
I was thinking that we might remove it soon anyways, but yeah good point

Reply:
hmm, I think ill leave the duplication here